### ISSUE TITLE
Create on a model without id

### ISSUE THREAD

**Issue by vbourgeois** [strongloop/loopback-connector-oracle#20]
Hello,

I created the model of a table that doesn't have primary key with the "discoverAndBuildModels" method of loopback-datasource-juggler.

When I tried to do a create with the generated model, I get the following error: 
TypeError: Cannot read property 'type' of undefined

At this line in oracle.js:
330     var type = this._models[model].properties[this.idName(model)].type;
Because "this.idName(model)" returns undefined.

Is there a way to do a create without a PK? Do I need to modify something in the generated model? 

Thanks.

**Comment by yellowred** [comment]
if `idInjection` is true loopback will generate an id for your model.
https://loopback.io/doc/en/lb2/Model-definition-JSON-file.html

**Comment by kaushaldeck** [comment]
I am working with a mongo db collection called users with the following fields : _id (default mongo created pk field), email, username, status.
I have created a model where I have marked email field as the id field. But after going through the full cycle when i run the get query for /users/{id} loopback queries only the _id field but doesnt query the email field (which I have marked as the id field in my model). Can someone help me in changing this? 

Thanks
